cmd/sti/main.go

```diff
@@ -100,57 +98,57 @@ func newCmdBuild(req *api.Request) *cobra.Command {
 			}()
 			// Attempt to restore the build command from the configuration file
 			if useConfig {
-				config.Restore(req, cmd)
+				config.Restore(cfg, cmd)
 			}
 
 			// If user specifies the arguments, then we override the stored ones
 			if len(args) >= 2 {
-				req.Source = args[0]
-				req.BaseImage = args[1]
+				cfg.Source = args[0]
+				cfg.BuilderImage = args[1]
 				if len(args) >= 3 {
-					req.Tag = args[2]
+					cfg.Tag = args[2]
 				}
 			}
 
-			if !validRequest(req) {
+			if len(validation.ValidateConfig(cfg)) != 0 {
```

Comment:
That was the initial goal, to have validation method available to be run in origin (see #144). But I agree we can revisit this method once again to do that. I've created an issue to track that #206. If you don't mind, I'll do that as a follow-up. I don't want to get this PR more hairy than it's already is :wink: